components/azure-iot/sdk_tests/tlsio_openssl_compact_ut/gballoc_ut_impl_2.h

```diff
@@ -0,0 +1,85 @@
+// Copyright(c) Microsoft.All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// This file is made an integral part of a unit test implementation with a #include. It
+// is broken out for reuse and readability. 
+
+#ifndef GBALLOC_UT_IMPL_2_H
+#define GBALLOC_UT_IMPL_2_H
+
+#include <stdbool.h>
+
+// This adds memory checking to the gballoc mocking
+// NOTE: Using ASSERTs wouuld be nice, but this file must preceed their definition
+// to avoid the malloc infinite recursion problem
+
+
+#ifndef GBALLOC_UT_IMPL_MAX_ALLOCS
+#define GBALLOC_UT_IMPL_MAX_ALLOCS 1000
+#endif
+
+static void* memory_blocks[GBALLOC_UT_IMPL_MAX_ALLOCS];
+static uint32_t memory_block_count;
+
+
+static void init_gballoc_checks()
+{
+    uint32_t i;
+	memory_block_count = 0;
```

Comment:
Note here and about the port layer in general: Different files seem to use different coding styles, in particular there seems to be a mix of spaces and tabs in some of them. Could you please check [ESP-IDF style guide](http://esp-idf.readthedocs.io/en/latest/contribute/style-guide.html) and update the code formatting accordingly? There are tools to format the code automatically in $IDF_PATH/tools directory.